include/zephyr/dsp/conversions.h

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_DSP_CONVERSIONS_H
+#define ZEPHYR_INCLUDE_ZEPHYR_DSP_CONVERSIONS_H
+
+#include <zephyr/dsp/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @ingroup math_dsp
+ * @defgroup math_conversions Basic Math Conversions
+ *
+ * Enables easy conversion of Q7, Q15, Q31, and float types.
+ * @{
+ */
+
+/**
+ * @brief Convert Q31 format to float
+ *
+ * @param[in]  src The Q31 input vector
+ * @param[out] dst The float output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_q31_to_float(const q31_t *src, float *dst, uint32_t count)
+{
+	for (uint32_t i = 0; i < count; ++i) {
+		dst[i] = (float)src[i] / (float)INT32_MAX;
+	}
+}
+
+/**
+ * @brief Convert Q15 format to float
+ *
+ * @param[in]  src The Q15 input vector
+ * @param[out] dst The float output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_q15_to_float(const q15_t *src, float *dst, uint32_t count)
+{
+	for (uint32_t i = 0; i < count; ++i) {
+		dst[i] = (float)src[i] / (float)INT16_MAX;
+	}
+}
+
+/**
+ * @brief Convert Q7 format to float
+ *
+ * @param[in]  src The Q7 input vector
+ * @param[out] dst The float output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_q7_to_float(const q7_t *src, float *dst, uint32_t count)
+{
+	for (uint32_t i = 0; i < count; ++i) {
+		dst[i] = (float)src[i] / (float)INT8_MAX;
+	}
+}
+
+/**
+ * @brief Convert floats to Q31 format
+ *
+ * @param[in]  src The float input vector
+ * @param[out] dst The Q31 output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_float_to_q31(const float *src, q31_t *dst, uint32_t count)
+{
+	for (uint32_t i = 0; i < count; ++i) {
+		dst[i] = (int32_t)(src[i] * (float)INT32_MAX);
+	}
+}
+
+/**
+ * @brief Convert floats to Q15 format
+ *
+ * @param[in]  src The float input vector
+ * @param[out] dst The Q15 output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_float_to_q15(const float *src, q15_t *dst, uint32_t count)
```

Comment:
Vice versa, shouldn't these return a shift value and/or possibly error if the float is larger than the q value can hold?

Reply:
Agree!

Reply:
I've made the change this morning but didn't finish writing the extra tests. I was originally thinking that the caller will manage this, but I'm ok with it being part of the API.